Sources/Commands/SwiftPackageTool.swift

```diff
@@ -315,15 +316,17 @@ extension SwiftPackageTool {
                 throw InternalError("Could not get the current working directory")
             }
             
-            if localFileSystem.exists(AbsolutePath(url)) {
-                let currentTemplateLocation = AbsolutePath(url)
-                let templateDirectory = configPath.appending(components: "templates", "new-package", url)
-                
-                guard localFileSystem.exists(currentTemplateLocation) && localFileSystem.isDirectory(currentTemplateLocation) else {
-                    throw InternalError("Template folder \(url) could not be found in: \(cwd)")
+            if let path = try? AbsolutePath(validating: url) {
+                if localFileSystem.exists(path) {
+                    let currentTemplateLocation = path
+                    let templateDirectory = configPath.appending(components: "templates", "new-package", url)
```

Comment:
maybe I am reading this wrong, but this seem odd? is it concatenating a full absolute path to the templates root dir?

Reply:
So this was originally a relative path to the cwd, but then you suggested that it should be an absolute path. So here url would just be an absolute path for the template folder. So no concatenation.

I think this should go back to being relative to the cwd.

Reply:
the way I see this being used is:

A user downloads a template to `/Users/me/Downloads/my-template` (as an alternative to cloning it)

then adds the template using 

`swift package add-template /Users/me/Downloads/my-template`

so URL in this case is an absolute path to `/Users/me/Downloads/my-template` and we would copy `/Users/me/Documents/my-template` into `~/.swiftpm/config/templates/new-package/my-template`

does this make sense? did you have something else in mind?